src/Microsoft.AspNet.Mvc.Razor/Compilation/CompilationFailedException.cs

```diff
@@ -24,49 +24,33 @@ public class CompilationFailedException : Exception
                 [NotNull] string filePath,
                 [NotNull] string fileContent,
                 [NotNull] string compiledContent,
-                [NotNull] IEnumerable<CompilationMessage> messages)
-            : base(FormatMessage(messages))
+                [NotNull] IEnumerable<IRuntimeCompilationMessage> messages)
+            : base(Resources.FormatCompilationFailed(filePath))
         {
-            FilePath = filePath;
-            FileContent = fileContent;
+            SourceFilePath = filePath;
+            SourceFileContent = fileContent;
             CompiledContent = compiledContent;
-            Messages = messages.ToList();
+            Messages = messages;
         }
 
         /// <summary>
         /// Gets the path of the Razor source file that produced the compilation failure.
         /// </summary>
-        public string FilePath { get; private set; }
+        public string SourceFilePath { get; private set; }
 
         /// <summary>
         /// Gets a sequence of <see cref="CompilationMessage"/> instances encountered during compilation.
         /// </summary>
-        public IEnumerable<CompilationMessage> Messages { get; private set; }
+        public IEnumerable<IRuntimeCompilationMessage> Messages { get; }
 
         /// <summary>
         /// Gets the content of the Razor source file.
         /// </summary>
-        public string FileContent { get; private set; }
+        public string SourceFileContent { get; private set; }
 
         /// <summary>
         /// Gets the generated C# content that was compiled.
         /// </summary>
         public string CompiledContent { get; private set; }
-
-        /// <inheritdoc />
-        public override string Message
-        {
-            get
-            {
-                return Resources.FormatCompilationFailed(FilePath) +
-                       Environment.NewLine +
-                       FormatMessage(Messages);
-            }
-        }
```

Comment:
The diagnostic page (and Helios) print the exception message at the top of the error page. This   error message would cause diagnostic messages to be printed twice over - once at the top and once as we enumerate through the errors. Having it repeated was redundant especially given there's plans for Helios to support the type.